### PR TITLE
Fixed running Sample (All Platforms)

### LINKED ISSUE #1441: [Bug] Samples.Android fails with FeatureNotSupportedException on Android 6
### Description

Stack trace:

```
Xamarin.Essentials.FeatureNotSupportedException: Specified method is not supported.
  at Xamarin.Essentials.AppActions.PlatformSetAsync (System.Collections.Generic.IEnumerable`1[T] actions) [0x0000d] in /Users/janus/github/Essentials/Xamarin.Essentials/AppActions/AppActions.android.cs:30
  at Xamarin.Essentials.AppActions.SetAsync (Xamarin.Essentials.AppAction[] actions) [0x00000] in /Users/janus/github/Essentials/Xamarin.Essentials/AppActions/AppActions.shared.cs:16
  at Samples.App.OnStart () [0x00087] in /Users/janus/github/Essentials/Samples/Samples/App.xaml.cs:51
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/archive-mono/2020-02/android/release/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1021
  at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <7d2292394f8c488b97f5bc2a0ac0240d>:0
  at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <7d2292394f8c488b97f5bc2a0ac0240d>:0
  at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00008] in <7d2292394f8c488b97f5bc2a0ac0240d>:0
  at at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.1(intptr,intptr)

```

### Steps to Reproduce

1. Open Xamarin.Essentials.sln.
2. Build the project Samples.Android.
3. Run it on an Android-6 device (API 23).

### Expected Behavior

The samples app runs without error.

### Actual Behavior

It fails with an exception, see above.

### Basic Information

- Version with issue: 1.6.0-pre2-1-g0dba844
- Last known good version: ?
- IDE: VS for Mac
- Platform Target Frameworks: 
  - Android: 10
- Affected Devices: Nexus 7

### PR DESCRIPTION
**Pull request by dimonovdd**
The app didn't run due to an "AppActions"

Fixes #1441

### CODE REVIEW

**Review by mattleibow — APPROVED** [review]
Thanks for this!

**Comment by janusw** [comment]
Note that there is an alternative (and more general) fix in #1442.

**Review by janusw — COMMENTED** [review]
This change is not general enough: It fixes the problem on MacOS and Tizen, but not on old Android versions (see #1441).

**Comment by jamesmontemagno** [comment]
This is the recommended route and just need to also check version numbers too before calling the API.

**Comment by janusw** [comment]
> This is the recommended route and just need to also check version numbers too before calling the API.

I really don't think it is a good idea to further extend the explicit check in this PR, since the logic becomes quite involved and it just duplicates what is already encoded in `AppActions.PlatformIsSupported`. I can see two reasonable solutions here:

1. try-catch:
```
    try {
        // do some AppAction stuff here
    } catch (FeatureNotSupportedException) {
    }
```
2. using a public 'IsSupported' method (this is PR #1442)
```
    if (AppActions.IsSupported) {
        // do some AppAction stuff here
    }
```

**Comment by dimonovdd** [comment]
> try {
>         // do some AppAction stuff here
>     } catch (FeatureNotSupportedException) {
>     }

@jamesmontemagno @mattleibow  @janusw 
I think this is the best fix for the problem in this case

File: Samples/Samples/App.xaml.cs [hunk 1]
```diff
@@ -48,19 +56,33 @@ protected override async void OnStart()
...
+                Debug.WriteLine($"{nameof(AppActions)} Exception: {ex}");
+            }
         }
 
         void AppActions_OnAppAction(object sender, AppActionEventArgs e)
         {
             // Don't handle events fired for old application instances
             // and cleanup the old instance's event handler
-            if (Application.Current != this && Application.Current is App app)
+            try
```
Comment (mattleibow) [inline Samples/Samples/App.xaml.cs:75]: This does not need to be wrapped in a try/catch as the actual registration check will catch it.

**Review by mattleibow — COMMENTED** [review]
This looks good and I think we can merge this before #1442 so that at least it works for now.

File: Samples/Samples/App.xaml.cs (hunk 1 above)
Comment (dimonovdd) [reply Samples/Samples/App.xaml.cs]: I must be tired.

**Review by mattleibow — APPROVED** [review]

**Comment by dimonovdd** [comment]
> /azp run
this failed. 

@mattleibow 
I don't have access to the logs. Is there anything I should fix?

**Review by jamesmontemagno — APPROVED** [review]

**Comment by mattleibow** [comment]
Why is CI not starting?

**Comment by mattleibow** [comment]
Thanks for this.

**Pull request merged**